### PR TITLE
Blockchain: Add tests for BIP32 M/0H/0/N addresses from seed

### PR DESCRIPTION
**Pull request by kmels**

### CODE REVIEW

File: src/api/Blockchain.cpp [hunk 1]
```diff
@@ -459,7 +459,7 @@ void Blockchain::init_path(
     switch (standard) {
         case BlockchainAccountType::BIP32: {
             path.add_child(
-                account | static_cast<std::uint32_t>(Bip32Child::HARDENED));
+                0 | static_cast<std::uint32_t>(Bip32Child::HARDENED));
```
Comment (justusranvier) [inline src/api/Blockchain.cpp]: This workaround is only valid for the first nym in the wallet.

The right way to fix this is to make sure account is initialized with a correct value.

File: tests/blockchain/Test_AllocateAddress.cpp [hunk 2]
```diff
@@ -84,12 +84,12 @@ TEST(Test_Blockchain, testAllocateAddress)
     .Blockchain().AllocateAddress(
                                   Identifier(Alice),
                                   Identifier(AliceAccountID),
-                                  label, BTC);
+                                  label, false);
```
Comment (justusranvier) [inline tests/blockchain/Test_AllocateAddress.cpp]: Use EXTERNAL_CHAIN here instead of false

File: tests/blockchain/main.cpp [hunk 3]
```diff
@@ -40,6 +40,7 @@
 #include "OTTestEnvironment.hpp"
 
 int main(int argc, char **argv) {
+  system("rm -r $HOME/.ot/");
```
Comment (justusranvier) [inline tests/blockchain/main.cpp]: This needs to be documented somewhere so that people know it will happen if they run the tests

Comment (kmels) [reply tests/blockchain/main.cpp]: Added a warning to the README for now.

**Pull request merged**